docs/decisions/0004-explicitly-declare-enterprise-for-role-assignments.rst

```diff
@@ -0,0 +1,59 @@
+Enterprise role assignments will record a customer UUID
+-------------------------------------------------------
+
+Status
+======
+
+Proposed - January 2021
+
+Context
+=======
+
+The ``SystemWideEnterpriseUserRoleAssignment`` model is used to grant role-based permissions
+to enterprise users.  These permissions generally control the users' access to data
+and ability to perform certain functions within an enterprise customer.  This assignment model does `not` currently
+refer to any specific customer - the "context" to which a role is applied is inferred from
+a user (an ``auth.User`` record) having been "linked" to a customer via the ``EnterpriseCustomerUser`` model.
+
+However, a single ``auth.User`` may be linked to multiple ``EnterpriseCustomers``, and if that user
+is assigned a role (such as ``enterprise_admin``), it is inferred that the role assignment applies
+to **every** customer to which the user is linked.  This is undesirable; we generally do not want an admin
+of `Customer A` to also have admin permissions for `Customer B` (to which the user may be linked as a mere ``learner``).
+Instead, the ``SystemWideEnterpriseUserRoleAssignment`` model should explicitly record *which* customer
+the assigned user has the given role in.
+
+Decisions
+=========
+
+Add an ``enterprise_customer`` field to the assignment model
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* It will be an optional field (for now?)...will ``null`` be a special/wildcard value?.
+* Can we ever make this a required field?  Specifically, will edX/internal users be problematic?
+
+We'll backfill the values of this field
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* It's fine to have multiple assignments for the ``enterprise_learner`` role on a single user -
+  what about ``enterprise_admin``?  Should we allow the backfilling process to create multiple user assignments
+  for this role and clean up the data afterwards?
```

Comment:
Can you say a bit more about the data cleanup?

Reply:
The data cleanup is the hand-audit I mentioned in my reply above.  For every user who is effectively an admin of two different enterprise customers, we'll have to manually determine (with help from support) which of the two customers they are actually an admin of, and which they are simply a learner within (unless they're truly an admin of both).